gradio/blocks.py

```diff
@@ -1910,6 +1909,10 @@ async def process_api(
         state_ids_to_track, hashed_values = self.get_state_ids_to_track(block_fn, state)
         changed_state_ids = []
 
+        from gradio.context import LocalContext
+
+        LocalContext.blocks.set(self)
```

Comment:
Interesting, why do we set it here and not in the init of Blocks?

Reply:
I thought this would be more "future proof" as it's the most up to date version of the Blocks before any sort of processing happens. But I don't think we need this actually.

Reply:
We need this to be done after init for `is_running` to be set. I think actually this is a good place for it.